### ISSUE TITLE
[BUG] Redis became unrepsponsive 

### ISSUE THREAD

**Issue by vzambo64** [RediSearch/RediSearch#4445]
The Redis stack server became unresponsive. This is not the first time it has happened the third time in 5 weeks of operation, so not reproducible.
The software watchdog was on and dumped a message (the delay setting was 2000msec):

```
6603:signal-handler (1707673004) 
--- WATCHDOG TIMER EXPIRED ---

------ STACK TRACE ------
EIP:
/lib/x86_64-linux-gnu/libpthread.so.0(pthread_rwlock_wrlock+0x1f4)[0x7fd61af06a74]

Backtrace:
/lib/x86_64-linux-gnu/libpthread.so.0(+0x12980)[0x7fd61af0c980]
/lib/x86_64-linux-gnu/libpthread.so.0(pthread_rwlock_wrlock+0x1f4)[0x7fd61af06a74]
/opt/redis-stack/lib/redisearch.so(RedisSearchCtx_LockSpecWrite+0x1b)[0x7fd618e133bb]
/opt/redis-stack/lib/redisearch.so(IndexSpec_DeleteDoc+0xab)[0x7fd618e25b3b]
/opt/redis-stack/lib/redisearch.so(Indexes_DeleteMatchingWithSchemaRules+0xca)[0x7fd618e2636a]
/opt/redis-stack/lib/redisearch.so(HashNotificationCallback+0x366)[0x7fd618dfbcb6]
/opt/redis-stack/bin/redis-server 127.0.0.1:6380(+0x187d34)[0x55a062628d34]
/opt/redis-stack/bin/redis-server 127.0.0.1:6380(notifyKeyspaceEvent+0x45)[0x55a06262db75]
/opt/redis-stack/bin/redis-server 127.0.0.1:6380(deleteExpiredKeyAndPropagate+0x4e)[0x55a0625cdc0e]
/opt/redis-stack/bin/redis-server 127.0.0.1:6380(expireIfNeeded+0xb2)[0x55a0625cdd52]
/opt/redis-stack/bin/redis-server 127.0.0.1:6380(lookupKey+0x66)[0x55a0625cde26]
/opt/redis-stack/bin/redis-server 127.0.0.1:6380(RM_OpenKey+0xb5)[0x55a0625e9d65]
/opt/redis-stack/lib/rejson.so(+0xdcdd1)[0x7fd619753dd1]
/opt/redis-stack/lib/rejson.so(+0xa8e2d)[0x7fd61971fe2d]
/opt/redis-stack/lib/redisearch.so(+0x1cc203)[0x7fd618e1b203]
/opt/redis-stack/lib/redisearch.so(+0x1ca8ca)[0x7fd618e198ca]
/opt/redis-stack/lib/redisearch.so(+0x1c9357)[0x7fd618e18357]
/opt/redis-stack/lib/redisearch.so(+0x1c9324)[0x7fd618e18324]
/opt/redis-stack/lib/redisearch.so(+0x1c9c8c)[0x7fd618e18c8c]
/opt/redis-stack/lib/redisearch.so(sendChunk+0xad)[0x7fd618dae67d]
/opt/redis-stack/lib/redisearch.so(AREQ_Execute+0x47)[0x7fd618daf1d7]
/opt/redis-stack/lib/redisearch.so(RSAggregateCommand+0x1c0)[0x7fd618dafdb0]
/opt/redis-stack/bin/redis-server 127.0.0.1:6380(RedisModuleCommandDispatcher+0x59)[0x55a062639cd9]
/opt/redis-stack/bin/redis-server 127.0.0.1:6380(call+0xfe)[0x55a0625a989e]
/opt/redis-stack/bin/redis-server 127.0.0.1:6380(processCommand+0x92a)[0x55a0625aa7da]
/opt/redis-stack/bin/redis-server 127.0.0.1:6380(processInputBuffer+0xeb)[0x55a062580acb]
/opt/redis-stack/bin/redis-server 127.0.0.1:6380(readQueryFromClient+0x3d1)[0x55a062581e81]
/opt/redis-stack/bin/redis-server 127.0.0.1:6380(+0x10b936)[0x55a0625ac936]
/opt/redis-stack/bin/redis-server 127.0.0.1:6380(+0x111249)[0x55a0625b2249]
/opt/redis-stack/bin/redis-server 127.0.0.1:6380(aeProcessEvents+0x321)[0x55a06252ce11]
/opt/redis-stack/bin/redis-server 127.0.0.1:6380(aeMain+0x1d)[0x55a06252d1fd]
/opt/redis-stack/bin/redis-server 127.0.0.1:6380(main+0x364)[0x55a0625227f4]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xe7)[0x7fd61ab2ac87]
/opt/redis-stack/bin/redis-server 127.0.0.1:6380(_start+0x2a)[0x55a062522eba]
6603:signal-handler (1707673004) --------
```

Reading the trace logs seems to me:
1. Aggregate request has been involved
2. Associated with JSON keys. We are using rejson to store objects and we have a lot of JSON indices.
3. Started to evict expired keys. Found a key to be deleted
4. The locking attempt resulted in deadlock. (the service is not responding for 19 hours)

Any idea how to solve the issue?

Environment:
 - GCP VM
 - OS: [ ubuntu 18.04.]
 - Redis: 7.2.2
 - ReJSON 2.6.7
 - search: 2.8.9
 - timeseries: 1.10.9
 - graph: compiled recently no version info

The last Aggregate requests without response:

ft.aggregate message:createdAt @pageId:{164542}@objectStatus:{active} TIMEOUT 2000 LOAD 3 '$ ' AS object SORTBY 2 @createdAt DESC MAX 10000 LIMIT 0 30

ft.aggregate userNameIndex @serviceId:{org}@objectStatus:{active} LOAD 3 '$.objectId' AS objectId

message:createdAt index:
 
[message_createdAt:index.txt](https://github.com/RediSearch/RediSearch/files/14237287/message_createdAt.index.txt)

userNameIndex
[userName_index.txt](https://github.com/RediSearch/RediSearch/files/14237303/userName_index.txt)

**Comment by GuyAv46** [comment]
Thanks for reaching out!
This is a known issue with indexing and searching JSONs on 2.8, it was fixed on 2.8.10.

**Comment by vzambo64** [comment]
Thank you - just got that issue.
